File: icms-web/src/services/notification.service.ts
import apiClient from './apiClient'

export type NotificationType = 'info' | 'success' | 'warning' | 'error' | 'work_order' | 'inventory' | 'maintenance' | 'quality'

export interface Notification {
  id: number
  type: NotificationType
  title: string
  message: string
  link?: string
  read: boolean
  created_at: string
}

export interface NotificationFilters {
  page?: number
  limit?: number
  unread_only?: boolean
  type?: NotificationType
}

class NotificationService {
  private baseUrl = '/api/v1/notifications'

  async getNotifications(filters: NotificationFilters = {}): Promise<{
    data: Notification[]
    total: number
    unread_count: number
  }> {
    const params = new URLSearchParams()
    if (filters.page) params.append('page', filters.page.toString())
    if (filters.limit) params.append('limit', filters.limit.toString())
    if (filters.unread_only) params.append('unread_only', 'true')
    if (filters.type) params.append('type', filters.type)
    return apiClient.get(`${this.baseUrl}?${params.toString()}`)
  }

  async markAsRead(id: number): Promise<void> {
    await apiClient.post(`${this.baseUrl}/${id}/read`)
  }

  async markAllAsRead(): Promise<void> {
    await apiClient.post(`${this.baseUrl}/read-all`)
  }

  async deleteNotification(id: number): Promise<void> {
    await apiClient.delete(`${this.baseUrl}/${id}`)
  }
}

export const notificationService = new NotificationService()

File: Backend/services/notification_service.py
from typing import Optional

from sqlalchemy.orm import Session

from models.notification import Notification


def create_notification(
    db: Session,
    user_id: int,
    notification_type: str,
    title: str,
    message: str,
    link: Optional[str] = None,
) -> Notification:
    notification = Notification(
        user_id=user_id,
        type=notification_type,
        title=title,
        message=message,
        link=link,
        read=False,
    )
    db.add(notification)
    return notification


def get_notifications(
    db: Session,
    user_id: int,
    skip: int = 0,
    limit: int = 50,
    unread_only: bool = False,
    notification_type: Optional[str] = None,
):
    query = db.query(Notification).filter(Notification.user_id == user_id)
    if unread_only:
        query = query.filter(Notification.read.is_(False))
    if notification_type:
        query = query.filter(Notification.type == notification_type)

    total = query.count()
    unread_count = db.query(Notification).filter(
        Notification.user_id == user_id,
        Notification.read.is_(False),
    ).count()
    data = query.order_by(Notification.created_at.desc()).offset(skip).limit(limit).all()
    return data, total, unread_count
